hackernews.el

```diff
@@ -545,11 +650,14 @@ rendered at the end of the hackernews buffer."
 ;;; Feeds
 
 ;;;###autoload
+(setq hackernews--visited-links-loaded nil)
 (defun hackernews (&optional n)
   "Read top N Hacker News stories.
 The Hacker News feed is determined by `hackernews-default-feed'
 and N defaults to `hackernews-items-per-page'."
   (interactive "P")
+  (if (not hackernews--visited-links-loaded)
+      (hackernews-visited-links-load))
```

Comment:
1. `hackernews--visited-links-loaded` shouldn't be necessary if we initialise the values of the `hackernews--visited-ids` alist to `nil` and later on replace them with hash tables.
2. The `hackernews` command is not the only entry point; there are also `hackernews-top-stories`, `hackernews-new-stories`, etc.

Given my comments above, I propose moving this logic to `hackernews--load-stories`:
```el
(defun hackernews--load-stories (feed n &optional append)
  "..."
  ;; Ensure visited links are set up
  (unless (cdar hackernews--visited-ids)
    (dolist (entry hackernews--visited-ids)
      (setcdr entry (make-hash-table)))
    (hackernews-visited-links-load)
    (add-hook 'kill-emacs-hook #'hackernews-save-visited-links))
  ...)
```